lib/r10k/action/deploy/environment.rb

```diff
@@ -43,6 +45,50 @@ def call
 
         private
 
+        def extract_credentials!
+          if @sshkey_path && @token_path
+            raise R10K::Error, "Cannot specify both an SSH key and a token to use with this deploy."
+          end
+
+          if @sshkey_path
+            if @sshkey_path == '-'
+              sshkey = $stdin.read
+              keypath = write_key_to_tmpfile(sshkey).path
+              { sshkey_file: keypath }
+            elsif File.exists?(@sshkey_path)
+              { sshkey_file: @sshkey_path }
+            else
+              raise R10K::Error, _("{%path} does not exist, cannot load SSH key") % { path: @sshkey_path }
+            end
+          elsif @token_path
+            if @token_path == '-'
+              token = $stdin.read
+            elsif File.exists?(@token_path)
+              token = File.read(@token_path).strip
+            else
+              raise R10K::Error, _("{%path} does not exist, cannot load OAuth token") % { path: @token_path }
+            end
+
+            unless valid_token?(token)
+              raise R10K::Error, _("Supplied token is not a valid OAuth token.")
```

Comment:
Nit: this is a little bit vague, because it could mean the the token is expired, too long, too short, etc. I think it should probably say something invalid characters found in the token, not that the token itself is invalid.

Reply:
Updated, lemme know what you think.